collector/sys_user_summary.go

```diff
@@ -0,0 +1,156 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const sysUserSummaryQuery = `
+	SELECT
+		user,
+		statements,
+		statement_latency,
+		table_scans,
+		file_ios,
+		file_io_latency,
+		current_connections,
+		total_connections,
+		unique_hosts,
+		current_memory,
+		total_memory_allocated
+	FROM
+		` + sysSchema + `.x$user_summary
+`
+
+var (
+	sysUserSummaryStatements = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "statements_total"),
+		" The total number of statements for the user",
+		[]string{"user"}, nil)
+	sysUserSummaryStatementLatency = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "statement_latency"),
+		"The total wait time of timed statements for the user",
+		[]string{"user"}, nil)
+	sysUserSummaryTableScans = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "table_scans_total"),
+		"The total number of table scans for the user",
+		[]string{"user"}, nil)
+	sysUserSummaryFileIOs = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "file_ios_total"),
+		"The total number of file I/O events for the user",
+		[]string{"user"}, nil)
+	sysUserSummaryFileIOLatency = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "file_io_seconds_total"),
+		"The total wait time of timed file I/O events for the user",
+		[]string{"user"}, nil)
+	sysUserSummaryCurrentConnections = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "current_connections"),
+		"The current number of connections for the user",
+		[]string{"user"}, nil)
+	sysUserSummaryTotalConnections = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "connections_total"),
+		"The total number of connections for the user",
+		[]string{"user"}, nil)
+	sysUserSummaryUniqueHosts = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "unique_hosts_total"),
+		"The number of distinct hosts from which connections for the user have originated",
+		[]string{"user"}, nil)
+	sysUserSummaryCurrentMemory = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "current_memory"),
+		"The current amount of allocated memory for the user",
+		[]string{"user"}, nil)
+	sysUserSummaryTotalMemoryAllocated = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, sysSchema, "memory_allocated_total"),
```

Comment:
Do you know what the unit is for these two metrics? I'm guessing it's bytes?

```suggestion
		prometheus.BuildFQName(namespace, sysSchema, "current_memory"),
		"The current amount of allocated memory for the user",
		[]string{"user"}, nil)
	sysUserSummaryTotalMemoryAllocated = prometheus.NewDesc(
		prometheus.BuildFQName(namespace, sysSchema, "memory_allocated_total"),
```
```suggestion
		prometheus.BuildFQName(namespace, sysSchema, "current_memory_bytes"),
		"The current amount of allocated memory for the user",
		[]string{"user"}, nil)
	sysUserSummaryTotalMemoryAllocated = prometheus.NewDesc(
		prometheus.BuildFQName(namespace, sysSchema, "memory_allocated_bytes_total"),
```

Reply:
Yeah, it's bytes. I have added the suffix in the metric names.